### ISSUE TITLE
BrowserWindows.setVisibleOnAllWorkspaces() does not work since 0.34.0

### ISSUE THREAD

**Issue by daxtero** [atom/electron#3302]
Since electron v.0.34.0 the Method BrowserWindow.setVisibleOnAllWorkspaces() does not work any more. The Window is visible always visible on all workspaces even if you call BrowserWindow.setVisibleOnAllWorkspaces(false).

**Comment by zcbenz** [comment]
Which operating system are you using?

**Comment by daxtero** [comment]
Thanks for fast answer!
It does not work on Mac OSX 10.11.1 (El Capitan), when trying it on Ubuntu 14.04.2 it works :)

**Comment by zcbenz** [comment]
I'm unable to reproduce on 10.10. Does `isVisibleOnAllWorkspaces()` return the correct value?

**Comment by daxtero** [comment]
```javascript
ipc.on('settings-appearance-widget-all-workspaces', (event, show) => {
    process.storage.set(show, C.ALLWORKSPACES, C.APPEARANCE);
    process.storage.save(false);
    console.log('before');
    console.log(process.windows.widget.isVisibleOnAllWorkspaces());
    process.windows.widget.setVisibleOnAllWorkspaces(show);
    console.log('after');
    console.log(process.windows.widget.isVisibleOnAllWorkspaces());
});
```

It will return correct values when switching between visible and non visible on all Workspaces, but it always show my widget on all Workspaces:
```javascript
before
false
after
true

before
true
after
false
```

**Comment by zcbenz** [comment]
Does it work if you call `require('remote').getCurrentWindow().setVisibleOnAllWorkspaces(true)` in the default app?

**Comment by daxtero** [comment]
It's the same behavior. Internally it's switching to false after calling require('remote').getCurrentWindow().setVisibleOnAllWorkspaces(false) in renderer process, but it's always shown on all workspaces.

**Comment by daxtero** [comment]
And it is the correct window ;) I logged window's id at creation and also logged console.log(require('remote').getCurrentWindow().id) at the time i switch visibleOnAllWorkspaces and it was the same :)

**Comment by zcbenz** [comment]
Can you provide a minimal example? I also tried on 10.11 but could not reproduce.

**Comment by zcbenz** [comment]
Closing until more information is provided.

**Comment by hariprasadiit** [comment]
Hi,

 for me, setting `setVisibleOnAllWorkspaces(true)` only works if i hide the dock icon on Mac OS using `app.dock.hide()` .

  is this the expected behaviour on mac?

**Comment by alt-jero** [comment]
Hi @hariprasadiit !

Thanks for the `app.dock.hide()` tip!

And now that we know that works, I have a theory as to why:
When an app has a dock icon, the app itself can be focussed. For example, when you click the app's dock icon, the name of the app shows up in the menu bar next to `File Edit View` etc. The same thing happens when you click in the app's window. When an app has no dock icon, you can click on the app's window and yet some other application will still have the menubar. So if it was `Finder` and you open and focus your app, the menubar still shows `Finder`.

TLDR: macOS (OS X) running apps show in the dock; hiding dock icon = background app thus app windows no longer linked to menubar and (more importantly) spaces.
